src/InfluxDB.LineProtocol/LineProtocolWriter.cs

```diff
@@ -0,0 +1,215 @@
+﻿using System;
+using System.Globalization;
+using System.IO;
+
+namespace InfluxDB.LineProtocol
+{
+    public class LineProtocolWriter
+    {
+        private static readonly DateTime UnixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        private readonly TextWriter textWriter;
+
+        public WriterPosition Position { get; private set; } = WriterPosition.StartOfLine;
+
+        public LineProtocolWriter() : this(new StringWriter())
+        {
+        }
+
+        internal LineProtocolWriter(TextWriter textWriter)
+        {
+            this.textWriter = textWriter ?? throw new ArgumentNullException(nameof(textWriter));
+        }
+
+        public LineProtocolWriter Measurement(string name)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentOutOfRangeException(nameof(name));
+            }
+
+            switch (Position)
+            {
+                case WriterPosition.StartOfLine:
+                    break;
+                case WriterPosition.FieldWriten:
+                case WriterPosition.TimestampWriten:
+                    textWriter.Write("\n");
```

Comment:
Do we expect the last item in a multiline payload to be followed by `\n`? (Not sure, myself.) If so, we might need a `.End()` method or similar to do this in that case.